src/NBitcoin/IBitcoinSerializable.cs

```diff
@@ -86,43 +85,62 @@ public static int GetSerializedSize(this IBitcoinSerializable serializable, Prot
             return GetSerializedSize(serializable, version, SerializationType.Disk);
         }
 
-        public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null)
+        public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
         {
-            network = network ?? Network.Main;
-
             var bms = new BitcoinStream(bytes)
             {
                 ProtocolVersion = version,
-                ConsensusFactory = network.Consensus.ConsensusFactory
+                ConsensusFactory = new DefaultConsensusFactory()
             };
+
             serializable.ReadWrite(bms);
         }
 
-        public static T Clone<T>(this T serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null) where T : IBitcoinSerializable, new()
+        public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ConsensusFactory consensusFactory, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
         {
-            network = network ?? Network.Main;
+            var bms = new BitcoinStream(bytes)
+            {
+                ProtocolVersion = version,
+                ConsensusFactory = consensusFactory
+            };
+
+            serializable.ReadWrite(bms);
+        }
 
-            T instance = network.Consensus.ConsensusFactory.TryCreateNew<T>();
+        public static T Clone<T>(this T serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION) where T : IBitcoinSerializable, new()
+        {
+            T instance = new DefaultConsensusFactory().TryCreateNew<T>();
             if (instance == null)
                 instance = new T();
 
-            instance.FromBytes(serializable.ToBytes(version, network), version, network);
+            instance.FromBytes(serializable.ToBytes(version), version);
 
             return instance;
         }
 
-        public static byte[] ToBytes(this IBitcoinSerializable serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, Network network = null)
+        public static T Clone<T>(this T serializable, ConsensusFactory consensusFactory, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION) where T : IBitcoinSerializable, new()
         {
-            network = network ?? Network.Main;
+            T instance = consensusFactory.TryCreateNew<T>();
```

Comment:
You don't need to create the instance this will happen inside the BitcoinStream.
`consensusFactory.TryCreateNew<T>() ` should only be called in the `BitcoinStream` unless we find a good reason not to.

Reply:
Actually @dangershony I'm going to remove this `Clone` method as its ony used when we clone the genesis block. We can use `Block.Load` for that now.